scipy/optimize/__init__.py

```diff
@@ -230,6 +230,7 @@
 
 from __future__ import division, print_function, absolute_import
 
+from .constraints import *
```

Comment:
To clarify: does it apply to all py files inside `optimize`? (because there are not public submodules in optimize). There are some without underscore, should they be renamed?